### PR TITLE
Rename and metrics

### LINKED ISSUE #93: Change all the geocoder names to service names
Rename the python library metrics methods to remove the geocoding word and use service instead: https://github.com/CartoDB/dataservices-api/blob/master/server/lib/python/cartodb_services/cartodb_services/metrics/quota.py#L32

### PR DESCRIPTION
**Pull request by ethervoid**
Closes #93, #99 

**Add the new config to chef**

- [x] Renamed the python library metrics functions
- [x] Create old version's folder to store the last versions
- [x] Refactor: Move redis and DB config logic to the python library
- [x] Generate the metrics log file (#99)

### CODE REVIEW

**Comment by ethervoid** [comment]
retest this please

**Comment by ethervoid** [comment]
@rafatower please check the PR and take it easy :_)

**Comment by ethervoid** [comment]
retest this please

**Comment by ethervoid** [comment]
retest this please

File: server/extension/Makefile [hunk 1]
```diff
@@ -5,27 +5,16 @@ EXTVERSION = $(shell grep default_version $(EXTENSION).control | sed -e "s/defau
 
 # The new version to be generated from templates
 NEW_EXTENSION_ARTIFACT = $(EXTENSION)--$(EXTVERSION).sql
+OLD_VERSIONS = $(wildcard old_versions/*.sql)
```
Comment (rafatower) [inline server/extension/Makefile]: playing with :fire:, you feel lucky today :smile:

Comment (ethervoid) [inline server/extension/Makefile]: I've tested it :)))

File: server/extension/cdb_dataservices_server--0.5.0--0.5.1.sql [hunk 2]
```diff
@@ -0,0 +1,571 @@
+DROP FUNCTION IF EXISTS cdb_dataservices_server._get_redis_conf_v2(text);
+DROP TYPE IF EXISTS cdb_dataservices_server._redis_conf_params;
+
+CREATE OR REPLACE FUNCTION cdb_dataservices_server._connect_to_redis(user_id text)
```
Comment (rafatower) [inline server/extension/cdb_dataservices_server--0.5.0--0.5.1.sql]: The problem here is that I don't have an easy/quick way of checking if this function is new or a modification over the existing one. In the later case I have no easy way of checking what that change actually was.

If I grep in the code for `_connect_to_redis` (or search for it in the long github page) I can simply die.

Sorry for the rant but it is an adequate time to rethink versioning and putting the cons on the tables hopefully will give us a better idea of how to improve things.

File: server/extension/sql/0.5.1/100_routing_helper.sql [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+  redis_conn = GD["redis_connection_{0}".format(username)]['redis_metrics_connection']
+  user_routing_config = GD["user_routing_config_{0}".format(username)]
+
+  quota_service = QuotaService(user_routing_config, redis_conn)
+
+  try:
+    client = MapzenRouting(user_routing_config.mapzen_app_key)
+
+    orig_lat = plpy.execute("SELECT ST_Y('%s') AS lat" % origin)[0]['lat']
```
Comment (rafatower) [inline server/extension/sql/0.5.1/100_routing_helper.sql]: (I wonder if there's a lighter way of doing this)

File: server/lib/python/cartodb_services/cartodb_services/metrics/config.py [hunk 4]
```diff
@@ -34,11 +35,11 @@ class RoutingConfig(ServiceConfig):
...
 
-    def __init__(self, redis_connection, username, orgname=None,
-                 mapzen_app_key=None):
+    def __init__(self, redis_connection, db_conn, username, orgname=None):
         super(RoutingConfig, self).__init__(redis_connection, username,
-                                             orgname)
-        self._mapzen_app_key = mapzen_app_key
+                                            orgname)
+        db_config = ServicesDBConfig(db_conn)
+        self._mapzen_app_key = db_config.mapzen_routing_app_key
```
Comment (rafatower) [inline server/lib/python/cartodb_services/cartodb_services/metrics/config.py]: is this configured in DB right now? cdb_conf table?

Comment (ethervoid) [inline server/lib/python/cartodb_services/cartodb_services/metrics/config.py]: Yes, is needed by the routing function

**Comment by rafatower** [comment]
My brain is too weak to do an effective review with such a PR. Sorry.

Please point me to any particular change you want me to review. Otherwise I'd need to resort to downloading your branch and diffing with kdiff3 or some other trick.

Also: this is the last PR we release this way. We need to have at the very least the SCM on our side.

**Comment by ethervoid** [comment]
Check the commits one by one (should be clearer) and avoid the release files. Don't check it because you're going to suffer a lot.

I agree with you, this kind of releases aren't what we should keep doing

**Comment by rafatower** [comment]
okay, I'm skipping the scaffolding commit and focusing on the others, expect further comments there.

don't take me wrong: we needed to test this in order to see the problems :smile:

**Comment by rafatower** [comment]
reviewed, I made some comments that are of your and santi's concern

**Pull request merged**